Sender.php

```diff
@@ -4,12 +4,14 @@
 
 namespace App;
 
-final class Sender
+abstract class Sender
 {
-    protected array $settings;
+    protected CredentialsDto $settings;
 
-    public function setCredentials(array $settings)
```

Comment:
Better use DTO objects because array type is more difficult to read and have more unpredictable field behaviors.